certbot-ci/certbot_integration_tests/certbot_tests/test_main.py

```diff
@@ -896,8 +898,50 @@ def test_preferred_chain(context: IntegrationTestsContext) -> None:
 
         dumped = misc.read_certificate(cert_path)
         assert 'Issuer: CN={}'.format(expected) in dumped, \
-               'Expected chain issuer to be {} when preferring {}'.format(expected, requested)
+            'Expected chain issuer to be {} when preferring {}'.format(expected, requested)
 
         with open(conf_path, 'r') as f:
             assert 'preferred_chain = {}'.format(requested) in f.read(), \
-                   'Expected preferred_chain to be set in renewal config'
+                'Expected preferred_chain to be set in renewal config'
+
+
+def test_certificate_validity(context: IntegrationTestsContext) -> None:
+    """Test that --certificate-validity is requesting certificates of the appropirate validity"""
+    if context.acme_server == 'boulder-v2':
+        pytest.skip('Boulder does not support NewOrder notAfter.')
+
+    context.certbot(['certonly', '--cert-name', 'newname', '-d', context.get_domain('newname'),
+                     '--certificate-validity', '86400'])
+
+    want_not_after = datetime.datetime.now(datetime.timezone.utc) + datetime.timedelta(days=1)
+
+    cert_path = join(context.config_dir, 'live', 'newname', 'cert.pem')
+
+    not_after_first = get_not_after(cert_path)
+    tc = unittest.TestCase()
+    tc.assertAlmostEqual(not_after_first, want_not_after, delta=datetime.timedelta(minutes=5))
+
+    context.certbot(['renew', '--force-renewal'])
```

Comment:
This function has a `force_renew=True` kwarg, so we can omit `--force-renewal`.

```suggestion
    context.certbot(['renew'])
```